src/components/Admin/ManageSensors/DeleteOldDataModal.tsx

```diff
@@ -55,14 +68,55 @@ function DeleteOldDataModal(): JSX.Element {
     setDownloadedDisclosureChecked(false);
     setUploadedDisclosureChecked(false);
     setIrreversibleDisclosureChecked(false);
+    setYear(0);
+    setMonth(0);
+    setDay(0);
+    setError('');
+
     onClose();
   }
 
+  /**
+   * Check that input dates are valid, sets error accordingly
+   */
+  const yearDigits = 4;
+  useEffect(() => {
+    const date = new Date(year, month - 1, day);
+    /**
+     * Determines whether a given date is old enough to serve as the delete before date.
+     *
+     * @returns Whether the inputted date is old enough to be the delete before date
+     */
+    const oldEnough = () => {
+      const offset = 7;
+      const sevenDaysAgo = new Date();
+      sevenDaysAgo.setDate(sevenDaysAgo.getDate() - offset);
```

Comment:
does this work as desired in the beginning of the month (i.e. on the 1st through 6th)?

Reply:
It should per the below screenshot (ignore me not knowing how to do math, the point still stands despite the variable names)
![image](https://user-images.githubusercontent.com/25434173/114488676-68980800-9bd7-11eb-8aab-2ae3a518e197.png)